commit 41fad83083a21800aafb08fbc888f05ae44d6c8a
Author: H.J. Lu <hjl.tools@gmail.com>
Date:   Fri Dec 19 13:49:58 2014 -0800

    Replace %ld with %jd and cast to intmax_t

diff --git a/libio/tst-ftell-active-handler.c b/libio/tst-ftell-active-handler.c
index 8549ccaafdf2623c..3d8030f32fcb3335 100644
--- a/libio/tst-ftell-active-handler.c
+++ b/libio/tst-ftell-active-handler.c
@@ -538,23 +538,23 @@ do_append_test (const char *filename)
 
   if (seek_ret != new_seek_ret)
     {
-      printf ("incorrectly modified file offset to %ld, should be %ld",
-	      new_seek_ret, seek_ret);
+      printf ("incorrectly modified file offset to %jd, should be %jd",
+	      (intmax_t)  new_seek_ret, (intmax_t) seek_ret);
       ret |= 1;
     }
   else
-    printf ("retained current file offset %ld", seek_ret);
+    printf ("retained current file offset %jd", (intmax_t) seek_ret);
 
   new_seek_ret = ftello (fp);
 
   if (seek_ret != new_seek_ret)
     {
-      printf (", ftello reported incorrect offset %ld, should be %ld\n",
-	      new_seek_ret, seek_ret);
+      printf (", ftello reported incorrect offset %jd, should be %jd\n",
+	      (intmax_t) new_seek_ret, (intmax_t) seek_ret);
       ret |= 1;
     }
   else
-    printf (", ftello reported correct offset %ld\n", seek_ret);
+    printf (", ftello reported correct offset %jd\n", (intmax_t) seek_ret);
 
   fclose (fp);
 
