commit 70996a373dd38296254029423ae6d87c02e7184a
Author: Roland McGrath <roland@hack.frob.com>
Date:   Fri Oct 24 12:37:36 2014 -0700

    Prototypify htonl and htons definitions.

diff --git a/inet/htonl.c b/inet/htonl.c
index 1d2d25f0498ba468..40145e55e0e7931f 100644
--- a/inet/htonl.c
+++ b/inet/htonl.c
@@ -21,8 +21,7 @@
 #undef	ntohl
 
 uint32_t
-htonl (x)
-     uint32_t x;
+htonl (uint32_t x)
 {
 #if BYTE_ORDER == BIG_ENDIAN
   return x;
diff --git a/inet/htons.c b/inet/htons.c
index 92cb4ea56786a91c..9651ebbd3a83fdbb 100644
--- a/inet/htons.c
+++ b/inet/htons.c
@@ -21,8 +21,7 @@
 #undef	ntohs
 
 uint16_t
-htons (x)
-     uint16_t x;
+htons (uint16_t x)
 {
 #if BYTE_ORDER == BIG_ENDIAN
   return x;
