commit a476ac4b45fe39b023bed55c852abad04d70c5df
Author: Siddhesh Poyarekar <siddhesh@redhat.com>
Date:   Fri Aug 1 14:30:43 2014 +0530

    Fix -Wundef warnings in regex_internal.h

diff --git a/include/libc-symbols.h b/include/libc-symbols.h
index fa0d590b0f072871..c555bf2939f2df7f 100644
--- a/include/libc-symbols.h
+++ b/include/libc-symbols.h
@@ -73,6 +73,10 @@
 #define STDC_HEADERS	1
 #define HAVE_MBSTATE_T	1
 #define HAVE_MBSRTOWCS	1
+#define HAVE_LIBINTL_H	1
+#define HAVE_WCTYPE_H	1
+#define HAVE_ISWCTYPE	1
+#define ENABLE_NLS	1
 
 /* The symbols in all the user (non-_) macros are C symbols.  */
 
diff --git a/posix/regex_internal.h b/posix/regex_internal.h
index b7f051eff00d1e7d..1e423caa128c1704 100644
--- a/posix/regex_internal.h
+++ b/posix/regex_internal.h
@@ -90,7 +90,7 @@
 # define SIZE_MAX ((size_t) -1)
 #endif
 
-#if (defined MB_CUR_MAX && HAVE_LOCALE_H && HAVE_WCTYPE_H && HAVE_WCHAR_H && HAVE_WCRTOMB && HAVE_MBRTOWC && HAVE_WCSCOLL) || _LIBC
+#if (defined MB_CUR_MAX && HAVE_WCTYPE_H && HAVE_ISWCTYPE) || _LIBC
 # define RE_ENABLE_I18N
 #endif
 
