commit 5cc45e102bdc19dec494e4ae8f0eb832f11af3e5
Author: Joseph Myers <joseph@codesourcery.com>
Date:   Fri Mar 8 16:47:43 2013 +0000

    Use ISO C prototype for __default_morecore.

diff --git a/malloc/morecore.c b/malloc/morecore.c
index 1e7b77749ff1700d..de013bda22ff6fb9 100644
--- a/malloc/morecore.c
+++ b/malloc/morecore.c
@@ -42,8 +42,7 @@ libc_hidden_proto (__sbrk)
    and return the start of data space, or NULL on errors.
    If INCREMENT is negative, shrink data space.  */
 __malloc_ptr_t
-__default_morecore (increment)
-     ptrdiff_t increment;
+__default_morecore (ptrdiff_t increment)
 {
   __malloc_ptr_t result = (__malloc_ptr_t) __sbrk (increment);
   if (result == (__malloc_ptr_t) -1)
