commit 80b5c505889ac4747aab54614d2bce18efa511ea
Author: Siddhesh Poyarekar <siddhesh@redhat.com>
Date:   Tue Jul 8 16:19:55 2014 +0530

    Fix Wundef warning for ELF_MACHINE_NO_REL on i386

diff --git a/sysdeps/i386/dl-machine.h b/sysdeps/i386/dl-machine.h
index 67d23228782d39c9..c52f17ae713fdebb 100644
--- a/sysdeps/i386/dl-machine.h
+++ b/sysdeps/i386/dl-machine.h
@@ -295,6 +295,7 @@ elf_machine_plt_value (struct link_map *map, const Elf32_Rel *reloc,
 /* The i386 never uses Elf32_Rela relocations for the dynamic linker.
    Prelinked libraries may use Elf32_Rela though.  */
 #define ELF_MACHINE_NO_RELA defined RTLD_BOOTSTRAP
+#define ELF_MACHINE_NO_REL 0
 
 #ifdef RESOLVE_MAP
 
